dot/network/service.go

```diff
@@ -550,15 +582,39 @@ func isInbound(stream libp2pnetwork.Stream) bool {
 	return stream.Stat().Direction == libp2pnetwork.DirInbound
 }
 
+func (s *Service) getMessageBuffer() [maxMessageSize]byte {
+	buf := s.bufPool.Get()
```

Comment:
You can directly do this.
```
return s.bufPool.Get().([maxMessageSize]byte)
```
`Sync.Pool` will never return nil since you have defined `New` for it and since we are using the sync pool for specifically storing `[maxMessageSize]byte`. Interface conversion will never panic.

Reply:
removed in favour of `sizedBufferPool`